src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -481,19 +483,24 @@ private AzureEnvironment MergeEnvironmentProperties(AzureEnvironment environment
                     }
                     else
                     {
-                        var subscriptions = subscriptionClient.Subscriptions.List().Subscriptions;
-                        if (subscriptions != null && subscriptions.Any())
+                        var subscriptions = (subscriptionClient.Subscriptions.List().Subscriptions ?? 
+                                                new List<Microsoft.Azure.Subscriptions.Models.Subscription>())
+                                            .Where(s => s.State.Equals("enabled", StringComparison.OrdinalIgnoreCase) || 
```

Comment:
May be safer to compare constant string to property in case server ever returns a null:
`"enabled".Equals(s.State, StringComparison.OrdinalIgnoreCase)`